tests/test_geometry.py

```diff
@@ -201,3 +201,57 @@ def test_geobox():
         assert abs(resolution[0]) > abs(geobox.extent.boundingbox.right - polygon.boundingbox.right)
         assert abs(resolution[1]) > abs(geobox.extent.boundingbox.top - polygon.boundingbox.top)
         assert abs(resolution[1]) > abs(geobox.extent.boundingbox.bottom - polygon.boundingbox.bottom)
+
+
+def test_wrap_dateline():
+    sinus_crs = geometry.CRS("""PROJCS["unnamed",
+GEOGCS["Unknown datum based upon the custom spheroid",
+DATUM["Not specified (based on custom spheroid)", SPHEROID["Custom spheroid",6371007.181,0]],
+PRIMEM["Greenwich",0],UNIT["degree",0.0174532925199433]],
+PROJECTION["Sinusoidal"],
+PARAMETER["longitude_of_center",0],
+PARAMETER["false_easting",0],
+PARAMETER["false_northing",0],
+UNIT["Meter",1]]""")
+    albers_crs = geometry.CRS('EPSG:3577')
+    geog_crs = geometry.CRS('EPSG:4326')
+
+    wrap = geometry.polygon([(12231455.716333, -5559752.598333),
+                             (12231455.716333, -4447802.078667),
+                             (13343406.236, -4447802.078667),
+                             (13343406.236, -5559752.598333),
+                             (12231455.716333, -5559752.598333)], crs=sinus_crs)
+    wrapped = wrap.to_crs(geog_crs)
+    assert wrapped.type == 'Polygon'
+    wrapped = wrap.to_crs(geog_crs, wrapdateline=True)
+    # assert wrapped.type == 'MultiPolygon' TODO: implement CutGeometryOnDateLineAndAddToMulti equivalent
```

Comment:
Nope, and probably never will be. It's quite wild to implement.
Hopefully, the GDAL method will be accessible through python API one day.
I'll add more details as a comment there I guess